src/Http/Session.php

```diff
@@ -220,8 +220,14 @@ public function __construct(array $config = [])
             'handler' => [],
         ];
 
-        if ($config['timeout']) {
-            $config['ini']['session.gc_maxlifetime'] = 60 * $config['timeout'];
+        if (isset($config['timeout']) && !isset($config['ini']['session.gc_maxlifetime'])) {
+            $maxlifetime = $config['timeout'] * 60;
+            if ($maxlifetime === 0) {
+                // If sessions are set to have no idle timeout, extend the
+                // gc_maxlifetime to 30 days so that sessions don't get reaped immediately
+                $maxlifetime = 60 * 60 * 24 * 30;
```

Comment:
Wonder if we should print a notice here instead of only hidden defaults so users are force to configure.

We'll be known as the 30-day session framework so we'll have to try to be explicit in the docs.

Reply:
I'm not sure this is the right solution either, but setting gc_maxlifetime to 0 is certainly wrong. We could check the current ini option value and use that as a default instead :shrug: That would at least remove the hidden fallback value.

Reply:
A notice would requires users to configure it through cake instead of directly in their php ini config which does feel incorrect.

Reply:
I pushed up another commit that I think will be safer. When `Session.timeout` is set > 0 the behavior is the same as now, but when `Session.timeout` is `0` then Cake won't change PHP configuration and users would get the maxlifetime defined in their PHP configuration.

Reply:
Looks ok. Basically, if you set a time out,  the gc will happen at the same time.